src/main/java/com/code_intelligence/jazzer/driver/FuzzTargetRunner.java

```diff
@@ -166,7 +170,21 @@ private static int runOne(long dataPtr, int dataLength) {
       //  1. Let libFuzzer pass in a null pointer if the byte array hasn't changed since the last
       //     call to our custom mutator and skip the read entirely.
       //  2. Implement a InputStream backed by Unsafe to avoid the copyToArray overhead.
-      mutator.read(new ByteArrayInputStream(copyToArray(dataPtr, dataLength)));
+      byte[] buf = copyToArray(dataPtr, dataLength);
+      boolean readCompletely = mutator.read(new ByteArrayInputStream(buf));
+
+      // Detect special libFuzzer inputs which can not be processed by the mutator framework.
+      boolean libFuzzerIndicator =
+          buf.length == 0 || (buf.length == 1 && buf[0] == LIBFUZZER_END_OF_CORPUS);
+
+      // If the mutator framework is used all constructed inputs are valid and read completely.
+      // Existing corpus files may not be valid for the current fuzz target anymore, though. In
+      // this case, print a warning once.
+      if (!(invalidCorpusFilesPresent || libFuzzerIndicator || readCompletely)) {
```

Comment:
`invalidCorpusFilesPresent` seems to be the flag that ensures we only warn once. Maybe call it `invalidCorpusFileWarningShown`? That still reads well in `invalidCorpusFilesPresent()`, but is less confusing here.

Reply:
Renamed.